packages/worker/src/core.ts

```diff
@@ -58,11 +73,46 @@ export class Core {
           : `http://${compiledServerOptions.address}`,
       },
       telemetryOptions,
+      logger: options.logger ?? new DefaultLogger('INFO'),
     };
     const native = await promisify(newCore)(compiledOptions);
+
     return new this(native, compiledOptions);
   }
 
+  private initLogPolling(options: CompiledCoreOptions, native: native.Core) {
+    this.shouldPollForLogs.next(true);
+
+    if (options.logger != null) {
+      const poll = promisify(corePollLogs);
+      this.logPollPromise = lastValueFrom(
+        of(this.shouldPollForLogs).pipe(
+          map((subject) => subject.getValue()),
+          takeWhile((shouldPoll) => shouldPoll),
+          concatMap(() => poll(native)),
+          map((logs) => {
+            for (const log of logs) {
+              if (log.level === 'TRACE') {
+                options.logger.trace(log.message);
+              } else if (log.level === 'DEBUG') {
+                options.logger.debug(log.message);
+              } else if (log.level === 'INFO') {
+                options.logger.info(log.message);
+              } else if (log.level === 'WARN') {
+                options.logger.warn(log.message);
+              } else if (log.level === 'ERROR') {
+                options.logger.error(log.message);
+              }
+            }
+          }),
+          delay(3), // Don't go wild polling as fast as possible
+          repeat()
+        )
+      );
+      this.logPollPromise.catch(() => undefined); // Prevent unhandled rejection
```

Comment:
If we're not exposing this promise we should at least *log* when this happens (yo dawg meme)